src/pubtools/sign/signers/msgsigner.py

```diff
@@ -431,10 +421,29 @@ def container_sign(self: MsgSigner, operation: ContainerSignOperation) -> Signin
             retries=self.retries,
             errors=errors,
         )
-        recvc.run()
-        errors = recvc.errors
+        recvt = RecvThread(recvc)
+        recvt.start()
 
-        if recvc.errors:
+        errors = SendClient(
+            messages=messages,
+            broker_urls=self.messaging_brokers,
+            cert=self.messaging_cert_key,
+            ca_cert=self.messaging_ca_cert,
+            retries=self.retries,
+            errors=errors,
+        ).run()
+
+        # wait for receiver to finish
+        recvt.join()
```

Comment:
Should it check errors of sender before waiting?

Reply:
fixed